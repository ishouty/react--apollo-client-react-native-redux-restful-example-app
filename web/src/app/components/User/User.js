import React, { Component } from 'react'
import { UsersText } from '../../constants/text/text'
import ProfileAvatarName from '../ProfileAvatarName/ProfileAvatarName'
import PropTypes from 'prop-types'
import { Row, Col } from 'react-bootstrap'

class User extends Component {
  render() {
    const { user, detail, displayAvatar } = this.props

    return (
      <Col
        className={
          detail ? 'user-container' : 'user-container-clickable'
        }
      >
        <Row>
          {displayAvatar && <ProfileAvatarName profile={user} />}
          {user.firstName && (
            <div>
              <i className="fa fa-user" aria-hidden="true"></i>{' '}
              {user.firstName} {user.lastName}
            </div>
          )}
          <h5>{UsersText.personalDetailsTitle}</h5>
          {user.email && (
            <div>
              <span>
                <i
                  className="fa fa-envelope-o"
                  aria-hidden="true"
                ></i>{' '}
                {UsersText.email}: {user.email}
              </span>
            </div>
          )}

          {user.userType && (
            <div>
              <span>
                <i className="fa fa-id-card" aria-hidden="true"></i>{' '}
                {UsersText.userType}: {user.userType}
              </span>
            </div>
          )}

          {user.cell && (
            <div>
              <span>
                <i className="fa fa-phone" aria-hidden="true"></i>{' '}
                {UsersText.tel}: {user.cell}
              </span>
            </div>
          )}
        </Row>
      </Col>
    )
  }
}

User.defaultProps = {
  displayAvatar: true
}

User.propTypes = {
  user: PropTypes.object.isRequired,
  detail: PropTypes.string,
  displayAvatar: PropTypes.bool
}

export default User
