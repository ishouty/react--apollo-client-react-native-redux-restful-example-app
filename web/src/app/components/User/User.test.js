import React, { Component } from 'React'
import { mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import User from './User'

describe('User', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it.only('should render the entire user profile', () => {
    const component = shallow(
      <User
        user={{
          name: {
            first: 'nguyen',
            last: 'nguyen'
          },
          firstName: 'andy',
          lastName: 'nguyen',
          email: 'admin@test.com',
          userType: 'admin',
          cell: '09393093093',
          picture: {
            picture: {
              medium: 'http://www.image.jpg'
            }
          }
        }}
      />
    )

    const componentHtml = component.html()

    expect(componentHtml).toMatchSnapshot()
    expect(componentHtml).toMatch(/andy/)
    expect(componentHtml).toMatch(/nguyen/)
    expect(componentHtml).toMatch(/admin@test.com/)
    expect(componentHtml).toMatch(/andy/)
    expect(componentHtml).toMatch(/09393093093/)
    expect(componentHtml).toMatch(/profile-container/)
  })

  it('should render the user profile without profile avatar', () => {
    const component = shallow(
      <User
        user={{
          firstName: 'andy',
          lastName: 'nguyen',
          email: 'admin@test.com',
          userType: 'admin',
          cell: '09393093093'
        }}
        displayAvatar={false}
      />
    )

    expect(component.find('.profile-container').exists()).toBe(false)
  })
})
