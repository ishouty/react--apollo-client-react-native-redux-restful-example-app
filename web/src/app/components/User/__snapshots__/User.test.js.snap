// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render the entire user profile 1`] = `"<div class=\\"user-container-clickable\\"><div class=\\"row\\"><div class=\\"inline\\"><div class=\\"profile-container\\"><div class=\\"inline\\"><div class=\\"profile-avatar-medium\\" style=\\"background-image:url(undefined)\\"></div></div><div class=\\"profile-name\\"><div class=\\"name\\">nguyen nguyen</div></div></div></div><div><i class=\\"fa fa-user\\" aria-hidden=\\"true\\"></i> andy nguyen</div><h5>Personal Details</h5><div><span><i class=\\"fa fa-envelope-o\\" aria-hidden=\\"true\\"></i> Email: admin@test.com</span></div><div><span><i class=\\"fa fa-id-card\\" aria-hidden=\\"true\\"></i> Role: admin</span></div><div><span><i class=\\"fa fa-phone\\" aria-hidden=\\"true\\"></i> Tel: 09393093093</span></div></div></div>"`;
